code/game/turfs/simulated/minerals.dm

```diff
@@ -59,7 +59,8 @@
 		to_chat(user, "<span class='notice'>You start picking...</span>")
 		P.playDigSound()
 
-		if(do_after(user,P.digspeed, target = src))
+		var/adjusted_digspeed = (user.getorganslot("dwarf_organ") ? (P.digspeed * 0.5) : P.digspeed)
```

Comment:
Still not happy with this.
Prefer some kind of proc on the mob, that by default returns 0, then you can do:
```
var/udigspeed = user.get_digspeed() //Eg: 0.5, 1, 2
var/adjusted_digspeed = udigspeed ? (P.digspeed * udigspeed) : P.digspeed
```